### ISSUE TITLE
--overlay option not working 

### ISSUE THREAD

**Issue by GodloveD** [apptainer/singularity#1896]
## Version of Singularity:

```
$ singularity --version
singularity version 3.0.0-alpha.1-117-gc297513f

$ uname -a
Linux ubuntu 4.15.0-29-generic #31-Ubuntu SMP Tue Jul 17 15:39:52 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

$ ls -l /dev/loop*
brw-rw---- 1 root disk  7,   0 Aug 29 10:12 /dev/loop0
brw-rw---- 1 root disk  7,   1 Aug 29 10:12 /dev/loop1
brw-rw---- 1 root disk  7,  10 Aug  7 11:52 /dev/loop10
brw-rw---- 1 root disk  7,  11 Aug  7 11:52 /dev/loop11
brw-rw---- 1 root disk  7,  12 Aug 23 10:04 /dev/loop12
brw-rw---- 1 root disk  7,  13 Aug 23 10:04 /dev/loop13
brw-rw---- 1 root disk  7,  14 Aug  7 11:52 /dev/loop14
brw-rw---- 1 root disk  7,  15 Aug  7 11:52 /dev/loop15
brw-rw---- 1 root disk  7,  16 Aug  7 11:52 /dev/loop16
brw-rw---- 1 root disk  7,  17 Aug  7 11:52 /dev/loop17
brw-rw---- 1 root disk  7,  18 Aug  7 11:52 /dev/loop18
brw-rw---- 1 root disk  7,  19 Aug  7 12:08 /dev/loop19
brw-rw---- 1 root disk  7,   2 Aug 29 10:12 /dev/loop2
brw-rw---- 1 root disk  7,  20 Aug 23 10:04 /dev/loop20
brw-rw---- 1 root disk  7,  21 Aug 23 10:04 /dev/loop21
brw-rw---- 1 root disk  7,  22 Aug 23 10:04 /dev/loop22
brw-rw---- 1 root disk  7,   3 Aug 29 10:12 /dev/loop3
brw-rw---- 1 root disk  7,   4 Aug 29 10:12 /dev/loop4
brw-rw---- 1 root disk  7,   5 Aug 23 10:04 /dev/loop5
brw-rw---- 1 root disk  7,   6 Aug  7 11:52 /dev/loop6
brw-rw---- 1 root disk  7,   7 Aug  7 11:52 /dev/loop7
brw-rw---- 1 root disk  7,   8 Aug  7 11:52 /dev/loop8
brw-rw---- 1 root disk  7,   9 Aug  7 11:52 /dev/loop9
crw-rw---- 1 root disk 10, 237 Aug  7 11:52 /dev/loop-control

```

### Expected behavior

```
$ singularity shell --overlay overlay my_container
Singularity: Invoking an interactive shell within container...

Singularity my_container:~/sandbox> exit
exit

$ sudo singularity shell --overlay overlay my_container
Singularity: Invoking an interactive shell within container...

Singularity my_container:~> exit
exit
```

### Actual behavior

```
$ singularity shell --overlay overlay my_container 
FATAL:   container creation failed: mount error: can't mount image /proc/self/fd/7: failed to find loop device: open /proc/self/fd/7: permission denied

$ sudo singularity shell --overlay overlay my_container 
FATAL:   container creation failed: mount error: can't mount image /proc/self/fd/7: failed to mount ext3 filesystem: invalid argument

```

### Steps to reproduce behavior

Try using overlays in 3.0

**Comment by GodloveD** [comment]
Just to further clarify, `overlay` above refers to an ext3 image I created using the `image create` command in 2.6.  

Also here is debug output from the 2 commands in question. 
```
$ singularity -d shell --overlay overlay my_container 
VERBOSE [U=0,P=121740]     singularity_message_level()               Set messagelevel to: 5
DEBUG   [U=0,P=121740]     init()                                    PIPE_EXEC_FD value: 7
VERBOSE [U=0,P=121740]     init()                                    Container runtime
VERBOSE [U=0,P=121740]     is_suid()                                 Check if we are running as setuid
DEBUG   [U=0,P=121740]     init()                                    Overlay seems supported by kernel
DEBUG   [U=0,P=121740]     init()                                    Drop privileges
DEBUG   [U=1000,P=121740]  init()                                    Read json configuration from pipe
DEBUG   [U=1000,P=121740]  init()                                    Set child signal mask
DEBUG   [U=1000,P=121740]  init()                                    Create socketpair for smaster communication channel
DEBUG   [U=1000,P=121740]  init()                                    Wait C and JSON runtime configuration from scontainer stage 1
DEBUG   [U=1000,P=121748]  set_parent_death_signal()                 Set parent death signal to 9
VERBOSE [U=1000,P=121748]  init()                                    Spawn scontainer stage 1
VERBOSE [U=1000,P=121748]  priv_escalate()                           Get root privileges
DEBUG   [U=0,P=121748]     set_parent_death_signal()                 Set parent death signal to 9
DEBUG   [U=0,P=121748]     prepare_scontainer_stage()                Entering in scontainer stage 1
DEBUG   [U=1000,P=121748]  set_parent_death_signal()                 Set parent death signal to 9
VERBOSE [U=1000,P=121748]  startup()                     Execute scontainer stage 1
DEBUG   [U=1000,P=121748]  SContainer()                  Entering scontainer stage 1
DEBUG   [U=1000,P=121740]  init()                                    Receiving configuration from scontainer stage 1
DEBUG   [U=1000,P=121740]  init()                                    Wait completion of scontainer stage1
VERBOSE [U=1000,P=121740]  priv_escalate()                           Get root privileges
VERBOSE [U=0,P=121740]     create_namespace()                        Create mount namespace
DEBUG   [U=0,P=121740]     init()                                    Create RPC socketpair for communication between scontainer and RPC server
VERBOSE [U=0,P=121740]     init()                                    Spawn smaster process
DEBUG   [U=0,P=121755]     set_parent_death_signal()                 Set parent death signal to 9
VERBOSE [U=0,P=121755]     init()                                    Spawn scontainer stage 2
VERBOSE [U=0,P=121755]     create_namespace()                        Create mount namespace
VERBOSE [U=0,P=121755]     init()                                    Spawn RPC server
VERBOSE [U=1000,P=121740]  startup()                     Execute smaster process
VERBOSE [U=0,P=121755]     startup()                     Serve RPC requests
DEBUG   [U=1000,P=121740]  setupSessionLayout()          Attempting to use overlayfs (enable overlay = try)
DEBUG   [U=1000,P=121740]  setupOverlayLayout()          Creating overlay SESSIONDIR layout
DEBUG   [U=1000,P=121740]  Init()                        Entering image format intializer
DEBUG   [U=1000,P=121740]  Init()                        sif format initializer returns: SIF magic not found
DEBUG   [U=1000,P=121740]  Init()                        sandbox format initializer returns: not a directory image
DEBUG   [U=1000,P=121740]  Init()                        squashfs format initializer returns: not a valid squashfs image
DEBUG   [U=1000,P=121740]  Init()                        Entering image format intializer
DEBUG   [U=1000,P=121740]  addRootfsMount()              Mounting block [squashfs] image: /home/david/sandbox/my_container
DEBUG   [U=1000,P=121740]  addKernelMount()              Checking configuration file for 'mount proc'
DEBUG   [U=1000,P=121740]  addKernelMount()              Adding proc to mount list
DEBUG   [U=1000,P=121740]  addKernelMount()              Checking configuration file for 'mount sys'
DEBUG   [U=1000,P=121740]  addKernelMount()              Adding sysfs to mount list
DEBUG   [U=1000,P=121740]  addDevMount()                 Checking configuration file for 'mount dev'
DEBUG   [U=1000,P=121740]  addDevMount()                 Adding dev to mount list
DEBUG   [U=1000,P=121740]  addHostMount()                Not mounting host file systems per configuration
VERBOSE [U=1000,P=121740]  addBindsMount()               Found 'bind path' = /etc/localtime, /etc/localtime
VERBOSE [U=1000,P=121740]  addBindsMount()               Found 'bind path' = /etc/hosts, /etc/hosts
DEBUG   [U=1000,P=121740]  isHomeAllowed()               Checking if user bind control is allowed
DEBUG   [U=1000,P=121740]  addHomeMount()                Adding home directory mount [/home/david:/home/david] to list using layer: overlay
DEBUG   [U=1000,P=121740]  isLayerEnabled()              Using Layer system: overlay
DEBUG   [U=1000,P=121740]  addTmpMount()                 Checking for 'mount tmp' in configuration file
DEBUG   [U=1000,P=121740]  addScratchMount()             Not mounting scratch directory: Not requested
DEBUG   [U=1000,P=121740]  create()                      Mount all
DEBUG   [U=1000,P=121740]  mountGeneric()                Mounting tmpfs to /usr/local/var/singularity/mnt/session
DEBUG   [U=1000,P=121740]  mountImage()                  Mounting loop device /dev/loop3 to /usr/local/var/singularity/mnt/session/rootfs
DEBUG   [U=1000,P=121740]  CleanupContainer()            Cleanup container
FATAL   [U=1000,P=121740]  SMaster()                     container creation failed: mount error: can't mount image /proc/self/fd/7: failed to find loop device: open /proc/self/fd/7: permission denied
```

And the sudo version:
```
$ sudo singularity -d shell --overlay overlay my_container 
[sudo] password for david: 
VERBOSE [U=0,P=121799]     singularity_message_level()               Set messagelevel to: 5
DEBUG   [U=0,P=121799]     init()                                    PIPE_EXEC_FD value: 7
VERBOSE [U=0,P=121799]     init()                                    Container runtime
VERBOSE [U=0,P=121799]     is_suid()                                 Check if we are running as setuid
DEBUG   [U=0,P=121799]     init()                                    Overlay seems supported by kernel
DEBUG   [U=0,P=121799]     init()                                    Read json configuration from pipe
DEBUG   [U=0,P=121799]     init()                                    Set child signal mask
DEBUG   [U=0,P=121799]     init()                                    Create socketpair for smaster communication channel
DEBUG   [U=0,P=121799]     init()                                    Wait C and JSON runtime configuration from scontainer stage 1
DEBUG   [U=0,P=121807]     set_parent_death_signal()                 Set parent death signal to 9
VERBOSE [U=0,P=121807]     init()                                    Spawn scontainer stage 1
VERBOSE [U=0,P=121807]     priv_escalate()                           Get root privileges
DEBUG   [U=0,P=121807]     set_parent_death_signal()                 Set parent death signal to 9
DEBUG   [U=0,P=121807]     prepare_scontainer_stage()                Entering in scontainer stage 1
VERBOSE [U=0,P=121807]     startup()                     Execute scontainer stage 1
DEBUG   [U=0,P=121807]     SContainer()                  Entering scontainer stage 1
DEBUG   [U=0,P=121799]     init()                                    Receiving configuration from scontainer stage 1
DEBUG   [U=0,P=121799]     init()                                    Wait completion of scontainer stage1
VERBOSE [U=0,P=121799]     priv_escalate()                           Get root privileges
VERBOSE [U=0,P=121799]     create_namespace()                        Create mount namespace
DEBUG   [U=0,P=121799]     init()                                    Create RPC socketpair for communication between scontainer and RPC server
VERBOSE [U=0,P=121799]     init()                                    Spawn smaster process
DEBUG   [U=0,P=121814]     set_parent_death_signal()                 Set parent death signal to 9
VERBOSE [U=0,P=121814]     init()                                    Spawn scontainer stage 2
VERBOSE [U=0,P=121814]     create_namespace()                        Create mount namespace
VERBOSE [U=0,P=121814]     init()                                    Spawn RPC server
VERBOSE [U=0,P=121799]     startup()                     Execute smaster process
VERBOSE [U=0,P=121814]     startup()                     Serve RPC requests
DEBUG   [U=0,P=121799]     setupSessionLayout()          Attempting to use overlayfs (enable overlay = try)
DEBUG   [U=0,P=121799]     setupOverlayLayout()          Creating overlay SESSIONDIR layout
DEBUG   [U=0,P=121799]     Init()                        Entering image format intializer
DEBUG   [U=0,P=121799]     Init()                        sif format initializer returns: SIF magic not found
DEBUG   [U=0,P=121799]     Init()                        sandbox format initializer returns: not a directory image
DEBUG   [U=0,P=121799]     Init()                        squashfs format initializer returns: not a valid squashfs image
DEBUG   [U=0,P=121799]     Init()                        Entering image format intializer
DEBUG   [U=0,P=121799]     addRootfsMount()              Mounting block [squashfs] image: /home/david/sandbox/my_container
DEBUG   [U=0,P=121799]     addKernelMount()              Checking configuration file for 'mount proc'
DEBUG   [U=0,P=121799]     addKernelMount()              Adding proc to mount list
DEBUG   [U=0,P=121799]     addKernelMount()              Checking configuration file for 'mount sys'
DEBUG   [U=0,P=121799]     addKernelMount()              Adding sysfs to mount list
DEBUG   [U=0,P=121799]     addDevMount()                 Checking configuration file for 'mount dev'
DEBUG   [U=0,P=121799]     addDevMount()                 Adding dev to mount list
DEBUG   [U=0,P=121799]     addHostMount()                Not mounting host file systems per configuration
VERBOSE [U=0,P=121799]     addBindsMount()               Found 'bind path' = /etc/localtime, /etc/localtime
VERBOSE [U=0,P=121799]     addBindsMount()               Found 'bind path' = /etc/hosts, /etc/hosts
DEBUG   [U=0,P=121799]     isHomeAllowed()               Checking if user bind control is allowed
DEBUG   [U=0,P=121799]     addHomeMount()                Adding home directory mount [/root:/root] to list using layer: overlay
DEBUG   [U=0,P=121799]     isLayerEnabled()              Using Layer system: overlay
DEBUG   [U=0,P=121799]     addTmpMount()                 Checking for 'mount tmp' in configuration file
DEBUG   [U=0,P=121799]     addScratchMount()             Not mounting scratch directory: Not requested
DEBUG   [U=0,P=121799]     create()                      Mount all
DEBUG   [U=0,P=121799]     mountGeneric()                Mounting tmpfs to /usr/local/var/singularity/mnt/session
DEBUG   [U=0,P=121799]     mountImage()                  Mounting loop device /dev/loop3 to /usr/local/var/singularity/mnt/session/rootfs
DEBUG   [U=0,P=121799]     mountImage()                  Mounting loop device /dev/loop4 to /usr/local/var/singularity/mnt/session/overlay-images/0
DEBUG   [U=0,P=121799]     CleanupContainer()            Cleanup container
FATAL   [U=0,P=121799]     SMaster()                     container creation failed: mount error: can't mount image /proc/self/fd/7: failed to mount ext3 filesystem: invalid argument
```

**Comment by cclerget** [comment]
FIxed by #1918
